verifier/src/lib.rs

```diff
@@ -219,14 +239,14 @@ mod tests {
     #[test]
     fn and_succeeds() {
         let and = And::new(AlwaysTrue, AlwaysTrue);
-        let verification = and.verify();
+        let verification = and.verify(NO_EVIDENCE);
         assert_eq!(verification.is_none().unwrap_u8(), 1);
```

Comment:
I realize this might be out of the scope of the PR and referring back to prior work, but why does the return type of `.verify()` use Option-like semantics rather than something more like a `Result<(), E>`? It's probable that there's a very good reason for this (for example, constant-time), but I figured I might as well double-check.

Reply:
> It's probable that there's a very good reason for this (for example, constant-time)
Correct it's for constant time.  Subtle doesn't have a Result type, probably because one would need to create the OK and the Error type to be constant time. So this is using CtOption<Error> as the result, where None (or 0) == success.
Up for ideas on how best to communicate this.
There is:
```rust
    /// In order to accommodate constant time operations this returns a
    /// [`CtOption`] instead of a [`Result`].
    fn verify(&self, evidence: &T) -> CtOption<Self::Error>;
```
On the `Verifier` trait, but perhaps it needs to be expanded more.